codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -341,7 +342,8 @@ public FlowControlledData(Http2Stream stream, ByteBuf buf, int padding, boolean
                                    ChannelPromise promise) {
             super(stream, padding, endOfStream, promise);
             queue = new CoalescingBufferQueue(promise.channel());
-            queue.add(buf, promise);
+            // Using false for the notifier as the may already fail the promise in error(...)
+            queue.add(buf, promise.isVoid() ? null : new ChannelPromiseNotifier(false, promise));
```

Comment:
Ah I see.  Can you point me to where the notifier would be created if we don't create it here?